From: Jeff Dike <jdike@addtoit.com>
Subject: [PATCH] vhost-net: fix vq_memory_access_ok error checking
References: FATE#311977
Git-commit: 179b284e2fc0c638035843968f7d7ab8ab701525
Patch-mainline: v2.6.34

vq_memory_access_ok needs to check whether mem == NULL

Signed-off-by: Jeff Dike <jdike@linux.intel.com>
Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 drivers/vhost/vhost.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: b/drivers/vhost/vhost.c
===================================================================
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -236,6 +236,10 @@ static int vq_memory_access_ok(void __us
 			       int log_all)
 {
 	int i;
+
+        if (!mem)
+                return 0;
+
 	for (i = 0; i < mem->nregions; ++i) {
 		struct vhost_memory_region *m = mem->regions + i;
 		unsigned long a = m->userspace_addr;
